app/assets/javascripts/admin/addon/components/admin-reports.gjs

```diff
@@ -13,22 +13,13 @@ import AdminSectionLandingWrapper from "admin/components/admin-section-landing-w
 
 export default class AdminReports extends Component {
   @service siteSettings;
-  @tracked reports = null;
+  @tracked reports;
   @tracked filter = "";
-  @tracked isLoading = true;
-
-  constructor() {
-    super(...arguments);
-    this.loadReports();
-  }
 
   @bind
-  loadReports() {
-    ajax("/admin/reports")
-      .then((json) => {
-        this.reports = json.reports;
-      })
-      .finally(() => (this.isLoading = false));
+  async loadReports() {
+    const response = await ajax("/admin/reports");
+    this.reports = response.reports;
```

Comment:
This kind of 'side effect' setting feels a bit weird. I think it would be cleaner to have `reports` only available inside `<AsyncContent`. We could also avoid the `bind` by using a getter. Something like:

```
@cached
get reportsPromise(){
  return ajax("/admin/reports").then((response) => response.reports);
}

<template>
  <AsyncContent @asyncData={{this.loadReportsPromise}}>
    <:content as |reports|>
      ... use value here
    </:content>
```

Would that work?

If we do need access to the reports in JS, then I think this would be a better case for direct usage of ember-async-data, rather than AsyncContent?

Reply:
Loading the reports was not issue. The problem was filtering them later. 

I solved it, using a function as helper.